### PR TITLE
[TT-4571] Implement security scheme conversions

### PR DESCRIPTION
**Pull request by furkansenharputlu**
This PR implements `securitySchemes` conversions of OAS <-> apidef.

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3902/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1878423523)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3902/merge
Commit: 911f0cd6b9aae1dc94b3377203d547894307e487 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1878455131)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3902/merge
Commit: ddf05916f6c9610b5f0693b543a9aa9b63257185 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1878485915)

File: apidef/oas/security.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+	token.Signature.Fill(authConfig)
+	if ShouldOmit(token.Signature) {
+		token.Signature = nil
+	}
+
+	if ShouldOmit(token) {
+		token = nil
+	}
+
+	s.GetTykExtension().Server.Authentication.SecuritySchemes[authConfig.Name] = token
```
Comment (jeffy-mathew) [inline apidef/oas/security.go:46]: can we use `getTykSecuritySchemes` here as well?

File: apidef/oas/authentication.go [hunk 2]
```diff
@@ -44,7 +44,8 @@ type Authentication struct {
 	GoPlugin *GoPlugin `bson:"goPlugin,omitempty" json:"goPlugin,omitempty"`
 	// CustomPlugin contains the configurations related to CustomPlugin authentication mode.
 	// Old API Definition: `auth_configs["coprocess"]`
-	CustomPlugin *CustomPlugin `bson:"customPlugin,omitempty" json:"customPlugin,omitempty"`
+	CustomPlugin    *CustomPlugin          `bson:"customPlugin,omitempty" json:"customPlugin,omitempty"`
+	SecuritySchemes map[string]interface{} `bson:"securitySchemes,omitempty" json:"securitySchemes,omitempty"`
```
Comment (jeffy-mathew) [inline apidef/oas/authentication.go:48]: With new `SecuritySchemes` we will only show `SecuritySchemes` while exporting. In that case, other auth types must be excluded from being marshalled, right?

File: apidef/oas/security.go [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+func (s *OAS) extractSecuritySchemeTo(ac *apidef.AuthConfig) {
+	if len(s.Security) == 0 {
+		return
+	}
+
+	for name := range s.Security[0] {
+		ref := s.Components.SecuritySchemes[name]
+		ac.Name = name
+		if ref.Value.Type == apiKey {
+			if ref.Value.In == header {
```
Comment (jeffy-mathew) [inline apidef/oas/security.go:167]: can we use a switch case here, makes it more readable

File: apidef/oas/security.go [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+	}
+
+	if !found {
+		s.Security.With(sec)
+	}
+
+	ref.Value.WithName(name).WithIn(loc).WithType(apiKey)
+}
+
+func (s *OAS) extractSecuritySchemeTo(ac *apidef.AuthConfig) {
```
Comment (jeffy-mathew) [inline apidef/oas/security.go:158]: `extractSecuritySchemeTo` seems like a generic name, will `extractApiKeyTo` be better?

Comment (furkansenharputlu) [reply apidef/oas/security.go:158]: Actually, I planned to be a generic one but it seems there will be no more cases where authconfig will be filled by security schemes.

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3902/merge
Commit: 992bce5f88cc6bded50272a4999d3301dbec65be 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1882190119)

**Review by jeffy-mathew — APPROVED** [review]

**Review by urbanishimwe — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/heads/master
Commit: 54ef28a564f6232f0f33c72f651febe289d2829c Implement security scheme conversions (#3902)
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1895624322)

**Pull request merged**